### PR TITLE
Fixes issue #546 - PopOverWidget doesn't work for satellite 6.9 settings page.

### PR DESCRIPTION
**Pull request by jameerpathan111**

### CODE REVIEW

File: airgun/widgets.py [hunk 1]
```diff
@@ -2124,44 +2124,43 @@ def fill(self, value):
...
 
-    ROOT = '.'
-    column_value = Text(".//span[contains(@class, 'editable-click')]")
-    header = Text(".//h3[contains(@class, 'popover-title')]")
-    input_box = TextInput(locator=".//input[contains(@class, 'form-control input-sm')]")
-    textarea = TextInput(locator=".//textarea[contains(@class, 'form-control input-large')]")
-    drop_down = Select(".//select[contains(@class,'form-control input-sm')]")
+    ROOT = "//div[contains(@class,'modal-content')]"
+    header = Text(".//h4[contains(@class, 'modal-title')]")
+    input_box = TextInput(locator=".//input[contains(@class, 'form-control')]")
```
Comment (omkarkhatavkar) [inline airgun/widgets.py:2155]: have you checked all the input box and sure we don't have the text area ?

File: airgun/widgets.py [hunk 2]
```diff
@@ -2124,44 +2124,43 @@ def fill(self, value):
...
+                    <button type="submit" class="btn btn-primary">
 
     Locator example::
 
-        //div[contains(@class, 'editable-open')]
+        //div[contains(@class,'modal-content')]
     """
 
-    ROOT = '.'
-    column_value = Text(".//span[contains(@class, 'editable-click')]")
```
Comment (tpapaioa) [inline airgun/widgets.py:2152]: What is the reason for removing `column_value` from `PopOverWidget` and adding it to `SettingsView`? Note that this breaks `PopOverWidget.read()`:

```
(Pdb) view.table.row()['Value'].widget.read()
*** AttributeError: 'PopOverWidget' object has no attribute 'column_value'
```
Can you instead keep `column_value` in `PopOverWidget` and implement a `click()` method in it if necessary?

**Review by tpapaioa — CHANGES_REQUESTED** [review]

File: airgun/widgets.py (hunk 2 above)
Comment (tpapaioa) [reply airgun/widgets.py:2152]: As a followup, I think I see the reason for moving `column_value`, as part of changing `PopOverWidget.ROOT` from `'.'` to `"//div[contains(@class,'modal-content')]"`. I think a better solution would be to keep the original `ROOT` and `column_value` in `PopOverWidget`, and move the widgets associated with the modal window that pops up to a separate view, e.g.,

```
class PopOverModalView(View):
    ROOT = "//div[contains(@class,'modal-content')]"
    header = Text(".//h4[contains(@class, 'modal-title')]")
    input_box = TextInput(locator=".//input[contains(@class, 'form-control')]")
    drop_down = Select(".//select[contains(@class,'form-control')]")
    submit = Text(".//button[@type='submit']")
```

`PopOverWidget.fill()` would then click on `column_value`, then instantiate `PopOverModalView` and access the modal's widgets through the view.

**Comment by latran** [comment]
@jameerpathan111 please address the necessary comments.  Afterwards, can we get some test result runs from usage of some of these changes?  And finally, if everything is satisfactory, please take it out of draft state so we can merge the changes.  There's many UI test case failures due to this issue.

File: airgun/widgets.py (hunk 2 above)
Comment (jameerpathan111) [reply airgun/widgets.py:2152]: @mshriver @omkarkhatavkar @tpapaioa @latran I have made changes as per this comment and it seems to be working. But I am facing new issue, when I try to update a setting new value is being appended to old one. 
Can someone take this up?

**Comment by tpapaioa** [comment]
Added a small fix to `wait_for_update()` in a separate commit.

**Comment by tpapaioa** [comment]
I was able to pass some tests with these commits. Running a full set of tests now, and I'll update the PR with the results as soon as they're ready.

**Comment by tpapaioa** [comment]
Settings-related tests passed, except for some errors that do not seem related to this issue.

```
# pytest tests/foreman/ui/test_settings.py
[...]
collected 16 items

tests/foreman/ui/test_settings.py ..E..F..FFssFssF.                      [100%]

==================================== ERRORS ====================================
____ ERROR at teardown of test_positive_httpd_proxy_url_update[http_proxy] _____
[...]
        yield setting_object
        setting_object.value = default_setting_value
>       setting_object.update({'value'})

pytest_fixtures/api_fixtures.py:562: 
[...]
        if http_error_msg:
>           raise HTTPError(http_error_msg, response=self)
E           requests.exceptions.HTTPError: 422 Client Error: Unprocessable Entity for url: https://satellite.example.com/api/v2/settings/http_proxy
[...]
=================================== FAILURES ===================================
_______ test_positive_host_dmi_uuid_duplicates[host_dmi_uuid_duplicates] _______
[...]
        with session:
            session.settings.update(f'name = {property_name}', property_dict[setting_update.name])
            result = session.settings.read(f'name = {property_name}')
>           assert result['table'][0]['Value'] == property_dict[setting_update.name]
E           AssertionError: assert '[]' == '[ hDpKfaFQlV ]'
E             - [ hDpKfaFQlV ]
E             + []

tests/foreman/ui/test_settings.py:193: AssertionError
___ test_positive_update_login_page_footer_text_with_long_string[login_text] ___
[...]
[etc.]
FAILED tests/foreman/ui/test_settings.py::test_positive_host_dmi_uuid_duplicates[host_dmi_uuid_duplicates]
FAILED tests/foreman/ui/test_settings.py::test_positive_update_login_page_footer_text_with_long_string[login_text]
FAILED tests/foreman/ui/test_settings.py::test_negative_settings_access_to_non_admin
FAILED tests/foreman/ui/test_settings.py::test_positive_update_email_delivery_method_sendmail
FAILED tests/foreman/ui/test_settings.py::test_negative_update_hostname_with_empty_fact[discovery_hostname]
ERROR tests/foreman/ui/test_settings.py::test_positive_httpd_proxy_url_update[http_proxy]
== 5 failed, 7 passed, 4 skipped, 62 warnings, 1 error in 2086.49s (0:34:46) ===
```

**Review by mshriver — APPROVED** [review]

**Review by tpapaioa — APPROVED** [review]

**Pull request merged**